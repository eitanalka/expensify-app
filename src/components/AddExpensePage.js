import React from 'react';

const AddExpensePage = () => (
  <div>
    This is from my add expense copmonent
  </div>
);

export default AddExpensePage;