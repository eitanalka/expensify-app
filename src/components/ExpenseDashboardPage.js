import React from 'react';

const ExpenseDashboardPage = () => (
  <div>
    This is from my dashboard copmonent
  </div>
);

export default ExpenseDashboardPage;