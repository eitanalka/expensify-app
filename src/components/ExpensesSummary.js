import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
  <div>
    <p>
      Viewing {props.expenses.length} expense{props.expenses.length !== 1 ? 's' : null} totalling {numeral(props.total / 100).format('$0,00.00')}
    </p>
  </div>
);

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters),
  total: selectExpensesTotal(state.expenses)
});

export default connect(mapStateToProps)(ExpensesSummary);