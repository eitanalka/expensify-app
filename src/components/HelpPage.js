import React from 'react';

const HelpPage = () => (
  <div>
    This is from my help copmonent
  </div>
);

export default HelpPage;