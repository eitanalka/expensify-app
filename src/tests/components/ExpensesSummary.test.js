import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} total={expenses[0].amount}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with 2 expenses', () => {
  const total = expenses[0].amount + expenses[1].amount + expenses[2].amount;
  const wrapper = shallow(<ExpensesSummary expenses={expenses} total={total}/>);
  expect(wrapper).toMatchSnapshot();
});