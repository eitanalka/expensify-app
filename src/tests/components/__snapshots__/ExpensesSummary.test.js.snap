// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense
     totalling 
    $1.95
  </p>
</div>
`;

exports[`should render ExpensesSummary with 2 expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expense
    s
     totalling 
    $1,141.95
  </p>
</div>
`;
